packages/knowledge/src/base.ts

```diff
@@ -469,7 +469,7 @@ export const base: ChartKnowledgeBase = {
       { minQty: 1, maxQty: 1, fieldConditions: ['Nominal', 'Ordinal'] },
       { minQty: 1, maxQty: 1, fieldConditions: ['Interval'] },
     ],
-    channel: ['ArcLength'],
+    channel: ['Angle', 'Color'],
```

Comment:
reason?

Reply:
I think the donut chart is almost identical to the pie except that it is hollow in the middle.
For example, in [echarts](https://echarts.apache.org/examples/zh/editor.html?c=pie-borderRadius), [vaga-lite](https://vega.github.io/vega-lite/examples/arc_donut.html), they both display the donut chart by setting the innerRadius (more like a style change).
In [G2Plot](https://g2plot.antv.vision/en/examples/case/statistical-scenario#pie), it uses angle and color field also.

Reply:
OIC, let me try to explain this... we set these props in CKB not by how the grammar of graphics defines the cases but how average users will understand the cases.

e.g. in a donut chart, users can't get the angle, actually. They may see the proportion via area, however, area is not accurate in donuts. So, when the inner and outer radius are getting close, in fact, it the length of arcs by which users get the difference.

Reply:
so, it should be [ 'ArcLength', 'Color' ] ?